openapi_python_client/parser/properties/__init__.py

```diff
@@ -388,9 +413,9 @@ def build_union_property(
     *, data: oai.Schema, name: str, required: bool, schemas: Schemas, parent_name: str
 ) -> Tuple[Union[UnionProperty, PropertyError], Schemas]:
     sub_properties: List[Property] = []
-    for sub_prop_data in chain(data.anyOf, data.oneOf):
+    for i, sub_prop_data in enumerate(chain(data.anyOf, data.oneOf)):
         sub_prop, schemas = property_from_data(
-            name=name, required=required, data=sub_prop_data, schemas=schemas, parent_name=parent_name
+            name=f"{name}_item{i}", required=required, data=sub_prop_data, schemas=schemas, parent_name=parent_name
```

Comment:
Personally "item" makes me think of a collection (e.g. list) and I think "type" might be more clear? But that's very subjective so up to you.

Reply:
Also, we should probably set `required=True` here. I think that would solve our extra `Unset` checks in the generated code. Really either the whole `Union` is required / nullable or None of it is. Not up to the individual types I don't think.

Would changing this cause any other problems you think?

Reply:
That sounds reasonable, but is it OK if we punt changing `required=True` to a future PR, to keep the scope of this PR limited, unless it's necessary here?